Source/Evolve/WarpXEvolve.cpp

```diff
@@ -568,30 +564,37 @@ WarpX::OneStep_multiJ (const amrex::Real cur_time)
         // Loop over multi-J depositions
         for (int i_depose = 0; i_depose < n_loop; i_depose++)
         {
-            // Move rho deposited previously, from new to old
-            PSATDMoveRhoNewToRhoOld();
-
             // Move J deposited previously, from new to old
-            // (when using assumption of J linear in time)
-            if (WarpX::J_linear_in_time) PSATDMoveJNewToJOld();
+            PSATDMoveJNewToJOld();
 
-            const amrex::Real t_depose = (WarpX::J_linear_in_time) ?
-                (i_depose-n_depose+1)*sub_dt : (i_depose-n_depose+0.5)*sub_dt;
+            const amrex::Real t_depose = (i_depose-n_depose+1)*sub_dt;
 
             // Deposit new J at relative time t_depose with time step dt
             // (dt[0] denotes the time step on mesh refinement level 0)
-            auto& current = (WarpX::do_current_centering) ? current_fp_nodal : current_fp;
             mypc->DepositCurrent(current, dt[0], t_depose);
             // Filter, exchange boundary, and interpolate across levels
             SyncCurrent();
             // Forward FFT of J
             PSATDForwardTransformJ();
 
-            // Deposit new rho
+            // Deposit mid rho and new rho
             if (WarpX::update_with_rho)
             {
-                // Deposit rho at relative time (i_depose-n_depose+1)*sub_dt
-                mypc->DepositCharge(rho_fp, (i_depose-n_depose+1)*sub_dt);
+                // Move rho deposited previously, from new to old
+                PSATDMoveRhoNewToRhoOld();
+
+                // Deposit rho at relative time t_depose - 0.5 * sub_dt
+                mypc->DepositCharge(rho_fp, t_depose - 0.5 * sub_dt);
+                // Filter, exchange boundary, and interpolate across levels
+                SyncRho();
+                // Forward FFT of rho_new
+                PSATDForwardTransformRho(0, 1);
+
+                // Move rho deposited previously, from new to mid
+                PSATDMoveRhoNewToRhoMid();
+
+                // Deposit rho at relative time t_depose
+                mypc->DepositCharge(rho_fp, t_depose);
```

Comment:
@RemiLehe Just to double check: Do the time values chosen here for the deposition of rho at half time ("mid rho") and at the final time ("new rho") look correct to you, as expressed in terms of the variables `t_depose` and `sub_dt`?